src/opuntia.erl

```diff
@@ -109,21 +122,16 @@ calculate(none, _, _) ->
     {none, 0};
 calculate(Shaper, 0, _) ->
     {Shaper, 0};
-calculate(#token_bucket_shaper{shape = {MaximumTokens, Rate},
+calculate(#token_bucket_shaper{shape = {MaximumTokens, Rate, TimeUnit},
                                available_tokens = LastAvailableTokens,
                                last_update = NativeLastUpdate,
-                               debt = LastDebt} = Shaper, TokensNowUsed, NativeNow) ->
-    NativeTimeSinceLastUpdate = NativeNow - NativeLastUpdate,
+                               debt = OverPenalisedInUnitsLastTime} = Shaper, TokensNowUsed, NativeNow) ->
 
-    %% This is now a float and so will all below be, to preserve best rounding errors possible
-    TimeSinceLastUpdate = convert_native_to_ms(NativeTimeSinceLastUpdate) + LastDebt,
+    TimeDiffInUnits = timediff_in_units(TimeUnit, NativeLastUpdate, NativeNow),
 
-    %% How much we might have recovered since last time
-    AvailableAtGrowthRate = Rate * TimeSinceLastUpdate,
-    UnboundedTokenGrowth = LastAvailableTokens + AvailableAtGrowthRate,
+    UnboundedTokenGrowth = unbounded_bucket_diff(Rate, LastAvailableTokens, TimeDiffInUnits),
```

Comment:
The only issue I have with the term `Growth` is that to me it suggests an increase, while it is the total amount.
I'd jsut call it `UnboundedTokens`, which would be consistent with `MaximumTokens`